File: deno/src/hrm/payslip/payslip.resolver.ts

/**
 * 发送企微工资条消息
 */
export async function sendMsgWxw(
  host: string,
  ids: string[],
) {
  const {
    sendMsgWxw,
  } = await import("./payslip.service.ts");
  
  const num = await sendMsgWxw(host, ids);
  return num;
}

/**
 * 一键发送企微工资条消息
 */
export async function sendMsgWxwOneKey(
  host: string,
) {
  const {
    sendMsgWxwOneKey,
  } = await import("./payslip.service.ts");
  
  const num = await sendMsgWxwOneKey(host);
  return num;
}

/**
 * 确认工资条
 */
export async function confirmPayslip(
  id: string,
) {
  const {
    confirmPayslip,
  } = await import("./payslip.service.ts");
  
  const num = await confirmPayslip(id);
  return num;
}

File: rust/src/src/hrm/payslip/payslip_resolver.rs
use anyhow::Result;

use crate::common::context::Ctx;

use super::payslip_service;

/// 发送企微工资条
pub async fn send_msg_wxw<'a>(
  ctx: &mut impl Ctx<'a>,
  host: String,
  ids: Vec<String>,
) -> Result<i32> {
  
  let res = payslip_service::send_msg_wxw(
    ctx,
    host,
    ids,
  ).await?;
  
  Ok(res)
}

/// 一键发送企微工资条
pub async fn send_msg_wxw_one_key<'a>(
  ctx: &mut impl Ctx<'a>,
  host: String,
) -> Result<i32> {
  
  let res = payslip_service::send_msg_wxw_one_key(
    ctx,
    host,
  ).await?;
  
  Ok(res)
}

/// 确认工资条
pub async fn confirm_payslip<'a>(
  ctx: &mut impl Ctx<'a>,
  id: String,
) -> Result<i32> {
  
  let res = payslip_service::confirm_payslip(
    ctx,
    id,
  ).await?;
  
  Ok(res)
}
